crypto/asn1/asn1_parse.c

```diff
@@ -132,22 +132,19 @@ static int asn1_parse2(BIO *bp, const unsigned char **pp, long length,
             const unsigned char *sp = p;
 
             ep = p + len;
-            if (BIO_write(bp, "\n", 1) <= 0)
+            if (BIO_write(bp, "\n", 1) < 0)
                 goto end;
             if (len > length) {
                 BIO_printf(bp, "length is greater than %ld\n", length);
-                ret = 0;
                 goto end;
             }
             if ((j == 0x21) && (len == 0)) {
                 for (;;) {
                     r = asn1_parse2(bp, &p, (long)(tot - p),
                                     offset + (p - *pp), depth + 1,
                                     indent, dump);
-                    if (r == 0) {
-                        ret = 0;
+                    if (r == 0)
```

Comment:
unsure if removing these ret = 0 will cause issues in other places.

Reply:
This is a further cleanup of the code:`ret = 0` is the default value anyway,
and it does not change unless the loop is exited.